### PR TITLE
[storage] make storage files _internal

### LINKED ISSUE #7799: [Storage-files] [Storage-queue]  Make client and model modules internal
Forces users to import from top level module

Important: document where it is supposed to be imported from

Also, include the version file

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves #7799

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/HISTORY.md [hunk 1]
```diff
@@ -9,6 +9,8 @@
 - Removed types that were accidentally exposed from two modules. Only `BlobServiceClient`, `ContainerClient`, 
 `BlobClient` and `LeaseClient` should be imported from azure.storage.blob.aio
 - `Logging` has been renamed to `BlobAnalyticsLogging`.
+- Client and model files have been renamed to be internal. Users should import from the top level modules:
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/HISTORY.md]: how about have been "made" internal?

File: sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py [hunk 2]
```diff
@@ -5,8 +5,16 @@
 # --------------------------------------------------------------------------
 # pylint: disable=no-self-use
 
+from typing import (  # pylint: disable=unused-import
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py]: typing isn't a part of python 2.7 or azure-core requirements. we must guard it.

File: sdk/storage/azure-storage-blob/HISTORY.md (hunk 1 above)
Comment (kristapratico) [reply sdk/storage/azure-storage-blob/HISTORY.md]: I'll rephrase. This change was made in files and queues as well.

**Pull request merged**